numba_dppy/dpnp_glue/dpnp_array_creations_impl.py

```diff
@@ -226,6 +225,82 @@ def dpnp_impl(a, b):
     return dpnp_impl
 
 
+@overload(stubs.dpnp.diag)
+def dpnp_diag_impl(v, k=0):
+    name = "diag"
+    dpnp_lowering.ensure_dpnp(name)
+
+    ret_type = types.void
+    """
+    dpnp source:
+    https://github.com/IntelPython/dpnp/blob/0.6.2/dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp#L67
+
+    Function declaration:
+    void dpnp_diag_c(
+        void* v_in, void* result1, const int k, size_t* shape, size_t* res_shape, const size_t ndim,
+        const size_t res_ndim)
+
+    """
+    sig = signature(
+        ret_type,
+        types.voidptr,
+        types.voidptr,
+        types.intp,
+        types.voidptr,
+        types.voidptr,
+        types.intp,
+        types.intp,
+    )
+    dpnp_func = dpnp_ext.dpnp_func("dpnp_" + name, [v.dtype.name, "NONE"], sig)
+
+    PRINT_DEBUG = dpnp_lowering.DEBUG
+
+    def dpnp_impl(v, k=0):
+        if v.size == 0:
+            raise ValueError("Passed Empty array")
+
+        if v.ndim != 1 and v.ndim != 2:
+            raise ValueError("Not supported")
+
+        if v.ndim == 1:
+            n = v.shape[0] + abs(k)
+
+            shape_result = (n, n)
+        else:
+            n = min(v.shape[0], v.shape[0] + k, v.shape[1], v.shape[1] - k)
+            if n < 0:
+                n = 0
```

Comment:
If `k > 0`, `v.shape[0] - k` will be the minimum. Again if `k < 0`, `v.shape[0] + k` will be the minimum. Therefore, why don't we do like:
```python
n = v.shape[0] - k if k > 0 else v.shape[0] + k  
```
or we can just do simply like
```python
p,q = v.shape[0] - k, v.shape[0] + k
n = (p if p > 0 else 0) if k > 0 else (q if q < 0 else 0)
```